packages/dataviews/src/view-list.tsx

```diff
@@ -192,6 +369,24 @@ export default function ViewList( props: ListViewProps ) {
 		defaultActiveId: getItemDomId( selectedItem ),
 	} );
 
+	// Manage focused item, when the active one is removed from the list.
+	const isActiveIdInList = store.useState(
+		( state: { items: any[]; activeId: any } ) =>
```

Comment:
I'm not sure how to provide the type for this. Happy to prepare a follow-up to improve it.

Reply:
Not sure exactly either, I guess this needs usage of ariakit types? 

Regardless, it's a bit weird that we have to use all these low level hooks (store and what not). I'm guessing the components API of ariakit is not easily usable for us here?

Reply:
> Regardless, it's a bit weird that we have to use all these low level hooks (store and what not). I'm guessing the components API of ariakit is not easily usable for us here?

Yeah. Each has its own reason:

- As per https://github.com/ariakit/ariakit/issues/3768 the store is a requisite for each `CompositeItem`
- As per https://github.com/WordPress/gutenberg/pull/60805#issuecomment-2108151014 we need to handle onkeydown for the "all actions" button to make sure arrow down doesn't open the menu but goes down the list. This is an issue with our Dropdown, as it works fine with Ariakit components.
- As per https://github.com/WordPress/gutenberg/pull/60805#issuecomment-2107561972 when deleting an element, we need to take care of updating `activeId` as ariakit doesn't do it for us.

Reply:
Thanks for the details.

> we need to handle onkeydown for the "all actions" button to make sure arrow down doesn't open the menu but goes down the list. This is an issue with our Dropdown, as it works fine with Ariakit components.

Maybe I'm not testing properly but for me it was actually opening the menu (and I felt that it was ok :P )

> As per https://github.com/WordPress/gutenberg/pull/60805#issuecomment-2107561972 when deleting an element, we need to take care of updating activeId as ariakit doesn't do it for us.

Couldn't that be considered a bug to be solved in ariakit?

Reply:
Just saw the ariakit issue :)

Reply:
> cc @WordPress/gutenberg-components for awareness. This PR implements two-dimensional arrow-key navigation with Ariakit's Composite & our own components. I'm leaving some comments for your awareness, in case we want to smooth it as an API. I plan to do similar work for other dataviews' layout in a few weeks.

I had to use the store's state in this PR, so I needed access to some types. I presume `any` can be improved by using Ariakit's types directly, but I also want to bring it up as another case where we may want to re-export from our components package.